agents-api/agents_api/queries/developers/create_developer.py

```diff
@@ -19,22 +24,25 @@
     settings
 )
 VALUES (
-    $1,
-    $2,
-    $3,
-    $4,
-    $5::jsonb
+    $1, -- developer_id
+    $2, -- email
+    $3, -- active
+    $4, -- tags
+    $5::jsonb -- settings
 )
 RETURNING *;
 """).sql(pretty=True)
 
 
-# @rewrap_exceptions(
-#     {
-#         QueryException: partialclass(HTTPException, status_code=403),
-#         ValidationError: partialclass(HTTPException, status_code=500),
-#     }
-# )
+@rewrap_exceptions(
+    {
+        asyncpg.UniqueViolationError: partialclass(
+            HTTPException,
+            status_code=404,
+            detail="The specified developer does not exist.",
```

Comment:
@Vedantsahai18  The error message seems a bit incorrect, as `UniqueViolationError` should raise when the entity already exists in the DB (correct me if I'm wrong), so it can be something like this: "Developer already exists."

Reply:
@whiterabbit1983 you are correct. On the other hand, I would prefer something like this
```py
 asyncpg.UniqueViolationError: lambda exc: HTTPException(
            status_code=409,
            detail=str(exc),
        )
```
This will give us more detailed description about the error. I have implemented it in the entries queries. Have a look

Reply:
@Vedantsahai18 we probably should not expose the whole exception to users, a better way is to show some readable message and log the error. The error code 409 is fine